ci/do_ci.sh

```diff
@@ -229,6 +229,19 @@ elif [[ "$1" == "cmake.c++20.test" ]]; then
   cmake ${CMAKE_OPTIONS[@]}  \
         -DCMAKE_CXX_FLAGS="-Werror $CXXFLAGS" \
         -DWITH_ASYNC_EXPORT_PREVIEW=ON \
+        -DWITH_STL=CXX20 \
```

Comment:
[`CXX_STANDARD`](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html) is a target property, not a environment variable in cmake.We can add `-DCMAKE_CXX_STANDARD=20` here to tell cmake to use c++20 by default.

Reply:
Are you saying the existing usage of `CXX_STANDARD` in the other (YAML) file is not correct? Or is there something that makes that prior usage there correct but the new usage there incorrect?

(I'm slightly confused because this comment thread seems to have nothing to do with the location it's attached to.)

Reply:
On [further inspection](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/ci/do_ci.sh#L87-L91) it looks like the CI script promotes the `CXX_STANDARD` environment variable to the flag `-DCMAKE_CXX_STANDARD=...`. It seems like the existing usage (and by extension, this new usage) is already correct.

Reply:
Sorry, I forgot we just set `CMAKE_CXX_STANDARD` by `CXX_STANRDARD` in `do_ci.sh` and also upgrade cmake by `CMAKE_VERSION`.Please just ignore my comment.